c10/core/SymInt.h

```diff
@@ -31,9 +29,52 @@ using SymIntNode = c10::intrusive_ptr<SymIntNodeImpl>;
 // named data_.
 class C10_API SymInt {
  public:
+  // TODO: this needs to only accept integers, not pointers
   /*implicit*/ SymInt(int64_t d) : data_(d){};
   SymInt() = default;
 
+  // TODO: these implementations are bad
```

Comment:
Why?

Reply:
They allocate a temporary object and then use the move constructor

Reply:
Could you update the comment to reflect this for future readers? :D